Improve look for student statistics page

kxmbrian opened the pull request.

weiqingtoh commented on app/views/course/statistics/_table.html.slim:
```diff
@@ -1,16 +1,28 @@
-     td = link_to student.experience_points,
-                  course_user_experience_points_records_path(current_course, student)
+     td.text-center
+       = index + 1
```
Sorry I didn't catch this earlier - actually in line 15 you can do `- students.each.with_index(1) do |student, index|` and you need to do `= index + 1` here. One of those cool ruby enumerable methods.

allenwq replied:
👍 TIL

weiqingtoh commented on app/services/course/student_statistics_service.rb:
```diff
@@ -18,11 +18,15 @@ def initialize(tutors)
+  # @return [Hash{Course::Group => Array<Course::GroupUser>}]
+  def group_managers_hash
+    @group_managers_hash ||=
+      @tutors.map(&:group_users).flatten.select(&:manager?).group_by(&:group)
```
Not sure whether it matters if we use `group_by(&:group)` vs `group_by(&:group_id)`?

kxmbrian replied:
`:group` seems to work, but will change it to `:group_id` to be safe.

allenwq replied:
Think it's ok, group id should also have much better performance.

weiqingtoh commented on app/services/course/student_statistics_service.rb:
```diff
@@ -18,11 +18,15 @@ def initialize(tutors)
+  # Maps groups to their managers
+  #
+  # @return [Hash{Course::Group => Array<Course::GroupUser>}]
+  def group_managers_hash
```
Should we make this private? Or doesn't matter?

kxmbrian replied:
This method is called in the view to decide if the Tutors column should be displayed.

kxmbrian commented:
Latest update has two changes:
- Added another method to the service and made `group_managers_hash` private.
- Removed the term 'tutor' form the service since the term can be ambiguous and the service might be used in a different context.

allenwq commented:
LGTM, @weiqingtoh any more comments ?
